src/search.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * @copyright Copyright (c) 2024 Fon E. Noel NFEBE <me@nfebe.com>
+ *
+ * @author Vincent Petry <me@nfebe.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import Vue from 'vue'
+
+import { getRequestToken } from '@nextcloud/auth'
+import { emit } from '@nextcloud/event-bus'
+import { translate, translatePlural } from '@nextcloud/l10n'
+import { generateFilePath } from '@nextcloud/router'
+
+import RoomSelector from './components/RoomSelector.vue'
```

Comment:
You can dynamically import it instead 
```
const RoomSelector = () => import('./components/RoomSelector.vue')
```

Reply:
If it could be done inside of init function

Reply:
Ah I assumed init won't be called as the script is loaded on `BeforeTemplateRenderedEvent`. For optimization purposes, if one of these workaround is possible it would be great:
- Lazy load the script itself (  when the unified search modal is opened) 
- Add the script immediately ( when Talk is active) and call the function that initializes the filter action when the unified search modal is opened (and have `RoomSelector` lazy loaded).

Reply:
You could refer to this commit for component lazy loading: https://github.com/nextcloud/spreed/pull/11589/commits/13de1cc65c6e8876ac57cbc29b4e62270e63c1ba

The benefit is that since you force this script to be loaded on every page where UnifiedSearch is present, we have to minimize its size (e.g. using dynamic import) so that the user doesn't load code every time that they won't use.